### PR TITLE
GAWB-2123: workspace tag apis

### PR DESCRIPTION
**Pull request by davidangb**
GET, PUT, PATCH, and DELETE apis for manipulating tags on an individual workspace.

Note that:
* in the api responses, the list of tags is sorted (case-insensitive).
* input strings are trimmed before saving tags

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/firecloud/Application.scala [hunk 1]
```diff
@@ -6,11 +6,11 @@ import org.broadinstitute.dsde.firecloud.dataaccess._
...
-                   val ontologyDAO: OntologyDAO,
-                   val rawlsDAO: RawlsDAO,
-                   val searchDAO: SearchDAO,
-                   val thurloeDAO: ThurloeDAO) {
+case class Application (agoraDAO: AgoraDAO,
+                   googleServicesDAO: GoogleServicesDAO,
+                   ontologyDAO: OntologyDAO,
+                   rawlsDAO: RawlsDAO,
+                   searchDAO: SearchDAO,
+                   thurloeDAO: ThurloeDAO) {
```
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/Application.scala:14]: changed this to a case class so I could use `copy` on it in unit tests.

**Review by rushtong — APPROVED** [review]
👍  Looks good, works good. 

With respect to the latest 'patch' commit, I don't understand how the patch test was working even though the actual function would add duplicates. The test should have been catching that case, but wasn't in practice.

**Comment by rushtong** [comment]
`WorkspaceApiServiceSpec` seems to be flaky.

**Review by rushtong — APPROVED** [review]
👍  To the latest changes. Going back to the original patch code, I can make the tests fail as they should.

**Pull request merged**